### PR TITLE
[2819] Update `ChannelList` docs

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2819

### 🎯 Goal

Update Compose `ChannelList` docs with new code snippets and screenshots where necessary.

### 🎨 UI Changes

screencapture-localhost-3000-chat-docs-sdk-android-compose-channel-components-channel-list-2021-12-30-09_59_11

**Component Previews:**
Screenshot 2021-12-30 at 09 55 38

Screenshot 2021-12-30 at 09 55 45

Screenshot 2021-12-30 at 09 55 50

### 🧪 Testing

1. Execute `npx stream-chat-docusaurus -i -s`
2. Navigate to a page with `ChannelList` component

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
~~- [ ] New code is covered by unit tests~~
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx [hunk 1]
```diff
@@ -55,15 +55,15 @@ fun ChannelList(
...
-* `onLastItemReached`: Handler called when reaching the last item in the list, to trigger pagination. You don't need to override this if you're using the default `viewModel`, but if you're using a custom one, you can add custom behavior.
-* `onChannelClick`: Handler for the user tapping on an item. Useful for starting the `MessagesScreen` in your app.
-* `onChannelLongClick`: Handler for the user long tapping on an item. By default, this updates state in the `viewModel`, which you can read to show custom UI and `Channel` actions, if you're using a custom `ViewModel` instance. Override if you're using the default `viewModel` and you want to change the behavior.
+* `viewModel`: The instance of the `ChannelListViewModel` that this component reads data from and sends events to. Pass in your own instance if you want more control over your business logic, such as changing `Channel` filters or sort order in runtime.
+* `onLastItemReached`: Handler for the user reaching the last item in the list to trigger pagination. You don't need to override this if you're using the default `viewModel`, but if you're using a custom one, you can add custom behavior.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx:64]: Maybe a slightly different wording here would be better.

- for when the user reaches
- called when the user reaches

File: docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx [hunk 2]
```diff
@@ -55,15 +55,15 @@ fun ChannelList(
...
+* `onChannelClick`: Handler when the user taps on an item. Useful for starting the `MessagesScreen`.
```
Comment (MarinTolic) [inline docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx:65]: Also here we might need a verb between handler and when

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [ ] ~~Bugs validated (bugfixes)~~
- [x] New feature tested and works **(previews)**
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx [hunk 3]
```diff
@@ -13,9 +13,9 @@ The **bound** version of the list uses the **stateless** list internally. That w
 
 Based on the provided state, the component shows the following UI:
 
-* `LoadingView`: If we're loading the initial data.
-* `EmptyView`: If loading is done but there is no data to show.
-* The list of channels: Channel items with various actions like item taps and long taps and pagination.
+* Loading view if we're loading the initial data.
```
Comment (filbabic) [inline docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx:16]: We shouldn't be using `View` in compose. Component or UI should be better to replace. Or in this case Indicator.

File: docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx (hunk 1 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx:64]: or `Handler when the user reaches`

File: docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx (hunk 2 above)
Comment (filbabic) [reply docusaurus/docs/Android/04-compose/03-channel-components/03-channel-list.mdx:65]: I think it's fine - it should be understandable

**Pull request merged**